crates/rpc-types-eth/src/transaction/receipt.rs

```diff
@@ -68,6 +68,74 @@ pub struct TransactionReceipt<T = ReceiptEnvelope<Log>> {
     pub contract_address: Option<Address>,
 }
 
+#[cfg(feature = "serde")]
+impl<'de, T> serde::Deserialize<'de> for TransactionReceipt<T>
+where
+    T: serde::Deserialize<'de>,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(serde::Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct ReceiptDeserHelper<T = ReceiptEnvelope<Log>> {
+            #[serde(flatten)]
+            inner: T,
+            transaction_hash: TxHash,
+            #[serde(default, with = "alloy_serde::quantity::opt")]
+            transaction_index: Option<u64>,
+            #[serde(default)]
+            block_hash: Option<BlockHash>,
+            #[serde(default, with = "alloy_serde::quantity::opt")]
+            block_number: Option<u64>,
+            #[serde(with = "alloy_serde::quantity")]
+            gas_used: u64,
+            // Optional fields for gas prices
+            // 1. Use effectiveGasPrice if present
+            // 2. Fallback to use gasPrice
+            // 3. Default to 0 if neither is present
+            #[serde(default = "default_gas", alias = "gasPrice", with = "alloy_serde::quantity")]
+            pub effective_gas_price: u128,
```

Comment:
this can also be private?
```suggestion
            effective_gas_price: u128,
```

Reply:
let's make this an option instead and use `Transaction::effective_gas_price` in case it's missing

Reply:
Made it an Option

wdym by using `Transaction::effective_gas_price`? `TransactionReceipt` doesn't implement `Transaction` trait